specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/computeRPCommon.json

```diff
@@ -0,0 +1,2027 @@
+          "description": "Specifies information about the image to use. You can specify information about platform images, marketplace images, or virtual machine images. This element is required when you want to use a platform image, marketplace image, or virtual machine image, but is not used in other creation operations."
+        },
+        "osDisk": {
+          "$ref": "#/definitions/OSDisk",
+          "description": "Specifies information about the operating system disk used by the virtual machine. For more information about disks, see [About disks and VHDs for Azure virtual machines](https://docs.microsoft.com/azure/virtual-machines/managed-disks-overview)."
+        },
+        "dataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataDisk"
+          },
+          "x-ms-identifiers": [
+            "lun"
+          ],
+          "description": "Specifies the parameters that are used to add a data disk to a virtual machine. For more information about disks, see [About disks and VHDs for Azure virtual machines](https://docs.microsoft.com/azure/virtual-machines/managed-disks-overview)."
+        },
+        "diskControllerType": {
+          "$ref": "#/definitions/DiskControllerType",
+          "description": "Specifies the disk controller type configured for the VM. **Note:** This property will be set to the default disk controller type if not specified provided virtual machine is being created with 'hyperVGeneration' set to V2 based on the capabilities of the operating system disk and VM size from the the specified minimum api version. You need to deallocate the VM before updating its disk controller type unless you are updating the VM size in the VM configuration which implicitly deallocates and reallocates the VM. Minimum api-version: 2022-08-01."
+        }
+      },
+      "description": "Specifies the storage settings for the virtual machine disks."
+    },
+    "UefiSettings": {
+      "properties": {
+        "secureBootEnabled": {
+          "type": "boolean",
+          "description": "Specifies whether secure boot should be enabled on the virtual machine. Minimum api-version: 2020-12-01."
+        },
+        "vTpmEnabled": {
+          "type": "boolean",
+          "description": "Specifies whether vTPM should be enabled on the virtual machine. Minimum api-version: 2020-12-01."
+        }
+      },
+      "description": "Specifies the security settings like secure boot and vTPM used while creating the virtual machine. Minimum api-version: 2020-12-01."
+    },
+    "EncryptionIdentity": {
+      "type": "object",
+      "properties": {
+        "userAssignedIdentityResourceId": {
+          "type": "string",
+          "description": "Specifies ARM Resource ID of one of the user identities associated with the VM."
+        }
+      },
+      "description": "Specifies the Managed Identity used by ADE to get access token for keyvault operations."
+    },
```

Comment:
This looks similar to patterns I have seen for CMK but maybe slightly different...

I'd like to understand the differences between this definition of encryption settings, and others like

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/customermanagedkeys.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/customermanagedkeys.json)

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/redisenterprise/resource-manager/Microsoft.Cache/stable/2023-11-01/redisenterprise.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/redisenterprise/resource-manager/Microsoft.Cache/stable/2023-11-01/redisenterprise.json)

Do you not support using a managed identity to access the encryption key?

Reply:
@pavanrachapudy90  could you please answer this question about EncryptionIdentity

Reply:
EncryptionIdentity property is for VM Azure Disk Encryption scenarios where ADE on host uses the encryption identity that is passed, to access customer's keyvault for disk related operations.

We do not support system assigned managed identity. Only user assigned identities. I believe the decision was made keeping pre-encrypted scenarios in mind where we cannot create a system assigned managed identity on VM creation workflow.

Reply:
That's pretty much it - system assigned identities are not supported, so only user assigned identities work in this scenario. Also, Azure Disk Encryption already exists and has it's metadata in a different format as it predates CMK by about 5 years. We decided on minimal changes here where we simply add the ability to select the identity to use for ADE rather than try to completely change ADE's properties to align with CMK.